examples/ui-demo/src/state/index.tsx

```diff
@@ -0,0 +1,170 @@
+"use client";
+
+import { alchemyConfig, Config, queryClient } from "@/app/config";
+import { AuthCardHeader } from "@/components/shared/AuthCardHeader";
+import { AlchemyClientState } from "@account-kit/core";
+import {
+  AlchemyAccountProvider,
+  AlchemyAccountsUIConfig,
+  AuthType,
+  useUiConfig,
+} from "@account-kit/react";
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useStore } from "zustand";
+import {
+  createDemoStore,
+  generateClassesForRoot,
+  generateStylesForRoot,
+} from "./store";
+
+export type ConfigContextType = {
+  config: Config;
+  setConfig: (partial: Partial<Config>) => void;
+  nftTransfered: boolean;
+  setNFTTransfered: Dispatch<SetStateAction<boolean>>;
+};
+
+export const ConfigContext = createContext<ConfigContextType | undefined>(
+  undefined
+);
+
+export function useConfig(): ConfigContextType {
+  const configContext = useContext(ConfigContext);
+
+  if (!configContext) {
+    throw new Error("config context must be present in root");
+  }
+
+  return configContext;
+}
+
+function convertDemoConfigToUiConfig(config: Config): AlchemyAccountsUIConfig {
+  const sections: AuthType[][] = [[{ type: "email" }]];
+
+  if (config.auth.showPasskey) {
+    sections.push([{ type: "passkey" }]);
+  }
+
+  if (config.auth.showSocial && config.auth.addGoogleAuth) {
+    if (config.auth.showPasskey) {
+      sections
+        .at(-1)!
+        .push({ type: "social", authProviderId: "google", mode: "popup" });
+    } else {
+      sections.push([
+        { type: "social", authProviderId: "google", mode: "popup" },
+      ]);
+    }
+  }
+
+  if (config.auth.showExternalWallets) {
+    sections.push([
+      {
+        type: "external_wallets",
+        walletConnect: { projectId: "30e7ffaff99063e68cc9870c105d905b" },
+      },
+    ]);
+  }
```

Comment:
In https://github.com/alchemyplatform/aa-sdk/pull/1039, I'm extracting the logic for building sections into a helper function because it needs to be reused between the ui and the code preview, so if my PR lands first you'll probably want to use that helper.

Reply:
it's a race! 🚗 

thanks for the heads up, will make sure all of that gets ported over if I end up in that state